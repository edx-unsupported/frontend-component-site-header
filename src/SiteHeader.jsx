import React from 'react';
import MediaQuery from 'react-responsive';

// Components
import DesktopHeader from './DesktopHeader';
import MobileHeader from './MobileHeader';


function SiteHeader(props) {
  return (
    <React.Fragment>
      <MediaQuery query="(max-width: 768px)">
        <MobileHeader {...props} />
      </MediaQuery>
      <MediaQuery query="(min-width: 769px)">
        <DesktopHeader {...props} />
      </MediaQuery>
    </React.Fragment>
  );
}


export default SiteHeader;
