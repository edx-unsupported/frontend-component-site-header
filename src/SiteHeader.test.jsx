
import React from 'react';
import { IntlProvider } from '@edx/frontend-i18n';
import TestRenderer from 'react-test-renderer';

import './__mocks__/reactResponsive.mock';

import SiteHeader from './index';

describe('<SiteHeader />', () => {
  const mainMenu = [
    {
      type: 'menu',
      href: '#',
      content: 'Courses',
      submenuContent: (
        <div>
          <a href="#link">link 1 </a>
          <a href="#link2">link 2 </a>
        </div>
      ),
    },
    {
      type: 'item',
      href: '#',
      content: 'Programs',
    },
    {
      type: 'item',
      href: '#',
      content: 'Schools & Partners',
    },
  ];

  const userMenu = [
    {
      type: 'item',
      href: '#',
      content: 'Dashboard',
    },
    {
      type: 'item',
      href: '#',
      content: 'Profile',
    },
    {
      type: 'item',
      href: '#',
      content: 'Account Settings',
    },
    {
      type: 'item',
      href: '#',
      content: 'Help',
    },
    {
      type: 'item',
      href: '#',
      content: 'Logout',
    },
  ];

  const loggedOutItems = [
    { type: 'item', href: '#', content: 'Login' },
    { type: 'item', href: '#', content: 'Sign Up' },
  ];

  it('renders correctly for desktop', () => {
    global.innerWidth = 1280;

    const component = (
      <IntlProvider locale="en" messages={{}}>
        <SiteHeader
          logo=""
          logoDestination="https://edx.org"
          logoAltText="edX"
          intl={{}}
          mainMenu={mainMenu}
          username="username"
          avatar={null}
          userMenu={userMenu}
          loggedOutItems={loggedOutItems}
        />
      </IntlProvider>
    );

    const wrapper = TestRenderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders correctly for mobile', () => {
    global.innerWidth = 500;

    const component = (
      <IntlProvider locale="en" messages={{}}>
        <SiteHeader
          logo=""
          logoDestination="https://edx.org"
          logoAltText="edX"
          intl={{}}
          mainMenu={mainMenu}
          username="username"
          avatar={null}
          userMenu={userMenu}
          loggedOutItems={loggedOutItems}
        />
      </IntlProvider>
    );

    const wrapper = TestRenderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
