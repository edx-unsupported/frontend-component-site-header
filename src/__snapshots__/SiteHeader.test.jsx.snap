// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteHeader /> renders correctly for desktop 1`] = `
<header
  className="site-header-desktop"
>
  <div
    className="container-fluid"
  >
    <div
      className="nav-container position-relative d-flex align-items-center"
    >
      <a
        className="logo"
        href="https://edx.org"
      >
        <img
          alt="edX"
          className="d-block"
          src=""
        />
      </a>
      <nav
        aria-label="Main"
        className="nav main-nav"
      >
        <div
          className="menu nav-item"
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup="menu"
            className="menu-trigger nav-link d-inline-flex align-items-center"
            href="#"
            onClick={[Function]}
          >
            Courses
             
            <IconMock
              aria-hidden={true}
              focusable="false"
              role="img"
            />
          </a>
        </div>
        <a
          className="nav-link"
          href="#"
        >
          Programs
        </a>
        <a
          className="nav-link"
          href="#"
        >
          Schools & Partners
        </a>
      </nav>
      <nav
        aria-label="Secondary"
        className="nav secondary-menu-container align-items-center ml-auto"
      >
        <a
          className="btn mr-2 btn-link"
          href="#"
        >
          Login
        </a>
        <a
          className="btn mr-2 btn-outline-primary"
          href="#"
        >
          Sign Up
        </a>
      </nav>
    </div>
  </div>
</header>
`;

exports[`<SiteHeader /> renders correctly for mobile 1`] = `
<header
  aria-label="Main"
  className="site-header-mobile d-flex justify-content-between align-items-center shadow sticky-top"
>
  <div
    className="w-100 d-flex justify-content-start"
  >
    <div
      className="menu position-static"
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Main Menu"
        className="menu-trigger icon-button"
        onClick={[Function]}
        title="Main Menu"
      >
        <IconMock
          aria-hidden={true}
          focusable="false"
          role="img"
          style={
            Object {
              "height": "1.5rem",
              "width": "1.5rem",
            }
          }
        />
      </button>
    </div>
  </div>
  <div
    className="w-100 d-flex justify-content-center"
  >
    <a
      className="logo"
      href="https://edx.org"
      itemType="http://schema.org/Organization"
    >
      <img
        alt="edX"
        className="d-block"
        src=""
      />
    </a>
  </div>
  <div
    className="w-100 d-flex justify-content-end align-items-center"
  >
    <nav
      aria-label="Secondary"
      className="menu position-static"
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Account Menu"
        className="menu-trigger icon-button"
        onClick={[Function]}
        title="Account Menu"
      >
        <span
          className="avatar overflow-hidden d-inline-flex rounded-circle null"
          style={
            Object {
              "height": "1.5rem",
              "width": "1.5rem",
            }
          }
        >
          <IconMock
            aria-hidden={true}
            className="text-muted"
            focusable="false"
            role="img"
            style={
              Object {
                "height": "1.5rem",
                "width": "1.5rem",
              }
            }
          />
        </span>
      </button>
    </nav>
  </div>
</header>
`;
